src/common/avatars/BaseAvatar.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import * as React from 'react';
+import styled from 'styled-components/native';
+import CachedImage from '../CachedImage';
+
+const StyledCachedImage = styled(CachedImage)`
+  border-radius: ${({ size }: Object): number => size / 2};
+  height: ${({ size }: Object): number => size};
+  width: ${({ size }: Object): number => size};
+`;
+
+const StyledTouchableOpacity = styled.TouchableOpacity`
+  background-color: #aaa;
+  border-radius: ${({ size }: Object): number => size / 2};
+  height: ${({ size }: Object): number => size};
+  width: ${({ size }: Object): number => size};
+`;
+
+export type BaseAvatarProps = {
+  imageRef?: React.Ref<typeof CachedImage>,
+  mutable?: boolean,
+  onPress?: () => void,
+  size: number,
+};
+
+type Props = BaseAvatarProps & { uri: string };
+
+class BaseAvatar extends React.PureComponent<Props> {
+  render() {
+    return (
+      <StyledTouchableOpacity
```

Comment:
What is this for?

Reply:
For onPress event and rounded circle the same size as the image, before the image loaded.